# Raw Data Description

In [1]:
# import relevant libraries

import pandas as pd
import seaborn as sns

import warnings

In [2]:
# set options

warnings.filterwarnings('ignore')
pd.set_option('display.max_columns', None)

There are two datasets for the CrossFit Open 2019. Both datasets are provided by [kaggle](https://www.kaggle.com/jeanmidev/crossfit-games). The origin of the data is from the official Reebok CrossFit website.The dataset "2019_opens_athletes.csv" contains information of all the athletes participated in the Open 2019 and in the dataset "2019_opens_scores.csv" there are the results of the five competition workouts.

In [3]:
# read the two datasets of 2019 - Open Athletes & Open Scores

df_19_ath = pd.read_csv('./data/2019_opens_athletes.csv')
df_19_sco = pd.read_csv('./data/2019_opens_scores.csv')

---

## Athlete Dataset

### dataset information

In [4]:
df_19_ath.head()

competitorid     competitorname firstname    lastname postcompstatus  \
0          2536    Samantha Briggs  Samantha      Briggs       accepted   
1        485089    Renata Pimentel    Renata    Pimentel       accepted   
2         16973    Carleen Mathews   Carleen     Mathews            NaN   
3        751083  Danila Capaccetti    Danila  Capaccetti            NaN   
4        313257        Hope Cicero      Hope      Cicero            NaN   

  gender          profilepics3key countryoforigincode countryoforiginname  \
0      F   0e63d-P2536_14-184.jpg                  GB      United Kingdom   
1      F  04e97-P485089_5-184.jpg                  BR              Brazil   
2      F   b663a-P16973_6-184.jpg                  US       United States   
3      F                pukie.png                  IT               Italy   
4      F  f204b-P313257_1-184.jpg                  US       United States   

   divisionid  affiliateid          affiliatename  age  height  weight  \
0          19         4098    CrossFit Black Five   37    1.70   61.23   
1          19        15868        CrossFit Gurkha   36    1.74   73.00   
2          19        10471  CrossFit Saint Helens   35    1.57   62.14   
3          19         9329   CrossFit Black Shark   35    1.70   71.00   
4          19          438      CrossFit Billings   36    1.55   61.23   

   overallrank  overallscore  is_scaled       division  
0            1            33          0  Women (35-39)  
1            2            66          0  Women (35-39)  
2            3           101          0  Women (35-39)  
3            4           139          0  Women (35-39)  
4            5           176          0  Women (35-39)

In [5]:
df_19_ath.shape

(572653, 19)

In [6]:
df_19_ath.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 572653 entries, 0 to 572652
Data columns (total 19 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   competitorid         572653 non-null  int64  
 1   competitorname       572653 non-null  object 
 2   firstname            572653 non-null  object 
 3   lastname             572653 non-null  object 
 4   postcompstatus       302 non-null     object 
 5   gender               572653 non-null  object 
 6   profilepics3key      572653 non-null  object 
 7   countryoforigincode  572348 non-null  object 
 8   countryoforiginname  572653 non-null  object 
 9   divisionid           572653 non-null  int64  
 10  affiliateid          572653 non-null  int64  
 11  affiliatename        540821 non-null  object 
 12  age                  572653 non-null  int64  
 13  height               304823 non-null  float64
 14  weight               323754 non-null  float64
 15  overallrank      

In [11]:
df_19_ath.describe()

competitorid     divisionid    affiliateid            age  \
count  5.726530e+05  572653.000000  572653.000000  572653.000000   
mean   1.109414e+06       8.567262   10433.175534      37.164978   
std    5.017513e+05       7.082857    6943.016568       9.012462   
min    8.600000e+01       1.000000       0.000000      16.000000   
25%    7.389000e+05       2.000000    4079.000000      31.000000   
50%    1.235059e+06       5.000000   10484.000000      37.000000   
75%    1.588160e+06      18.000000   16626.000000      43.000000   
max    1.715660e+06      19.000000   23323.000000     125.000000   

              height         weight    overallrank   overallscore  \
count  304823.000000  323754.000000  572653.000000  572653.000000   
mean        1.743674      78.343412   56143.535611  269979.915326   
std         1.439308      44.720039   55319.364617  245739.184643   
min         0.010000   -9054.150000       1.000000       9.000000   
25%         1.670000      67.000000    9055.000000   49581.000000   
50%         1.750000      78.930000   31697.000000  175493.000000   
75%         1.800000      88.450000   99381.000000  480742.000000   
max       266.240000   16960.000000  185551.000000  792781.000000   

           is_scaled  
count  572653.000000  
mean        0.115452  
std         0.319567  
min         0.000000  
25%         0.000000  
50%         0.000000  
75%         0.000000  
max         1.000000

* 572,653 entries in the dataset
* 19 features: 8 numeric and 11 categorical
* 10,3 % missing cells
* no duplicate rows
* memory size 380 MB

### feature explanation

* competitorid: each athlete is identified by the unique id (competitorid can be found in the score dataset too)
* competitorname: the full name of an athlete containing both first name and last name
* firstname: the first name of an athlete
* lastname: the last name of an athlete
* postcompstatus: ???
* gender: gender of an athlete (male, female)
* profilepics3key: name of the picture in athlete profile
* countryoforigincode: the abbreviated code of an athlete's country of origin
* countryoforiginname: full name of an athlete's country of origin
* divisionid: ???
* affiliateid: the unique id of an athlete's affiliate
* affiliatename: the official name of an athlete's affiliate
* age: the age of an athlete
* height: the height of an athlete (in meters)
* weight: the weight of an athlete (in kg)
* overallrank: the rank of an athlete considering all five competition workouts
* overallscore: the score an athlete has reached after all five workouts
* is_scaled: displays if an athlete belongs to a scaling division (overall)
* division: the division (grouped by age) an athlete belongs to

### neglected features

* competitorname: contains the information of firstname and lastname
* postcompstatus: almost entire data is missing
* profilepics3key: just the name of a picture
* countryoforiginname: contains same information as countryoforigincode
* dividionid: information of age-grouped divisions already in division

---

## Score Dataset

### dataset information

In [8]:
df_19_sco.head()

affiliate                                          breakdown  \
0  CrossFit RDU                   9 rounds +\n10 wall-ball shots\n   
1  CrossFit RDU  Within 16 minutes:\n3 rounds +\n25 toes-to-bar...   
2  CrossFit RDU  200-ft. OH lunge\n50 box step-ups\n50 strict H...   
3  CrossFit RDU                                 132 reps\n6 rounds   
4  CrossFit RDU                                           210 reps   

   competitorid     division  is_scaled           judge  ordinal  rank  \
0         96511  Men (45-49)          0     Erin Miller        1     1   
1         96511  Men (45-49)          0  Harper Thorsen        2     4   
2         96511  Men (45-49)          0  Harper Thorsen        3     1   
3         96511  Men (45-49)          0  Harper Thorsen        4    36   
4         96511  Men (45-49)          0  Harper Thorsen        5     1   

   scaled     score scoredisplay       scoreidentifier   time  
0       0  13520000     352 reps  27f30f9a8c0a564ae799    NaN  
1       0  13420368     342 reps  0ed3d1264f25a8f1890d    NaN  
2       0  11800018         9:42  f2a143399a330c95321b  582.0  
3       0  11320009        11:51  89101e401c6c85997363  711.0  
4       0  12100573        10:27  f7588c9174f1fe90f5c4  627.0

In [9]:
df_19_sco.shape

(2863265, 13)

In [10]:
df_19_sco.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2863265 entries, 0 to 2863264
Data columns (total 13 columns):
 #   Column           Dtype  
---  ------           -----  
 0   affiliate        object 
 1   breakdown        object 
 2   competitorid     int64  
 3   division         object 
 4   is_scaled        int64  
 5   judge            object 
 6   ordinal          int64  
 7   rank             int64  
 8   scaled           int64  
 9   score            int64  
 10  scoredisplay     object 
 11  scoreidentifier  object 
 12  time             float64
dtypes: float64(1), int64(6), object(6)
memory usage: 284.0+ MB


In [12]:
df_19_sco.describe()

competitorid     is_scaled       ordinal          rank        scaled  \
count  2.863265e+06  2.863265e+06  2.863265e+06  2.863265e+06  2.863265e+06   
mean   1.109414e+06  1.154521e-01  3.000000e+00  5.399598e+04  2.754048e-01   
std    5.017510e+05  3.195668e-01  1.414214e+00  5.206022e+04  4.467181e-01   
min    8.600000e+01  0.000000e+00  1.000000e+00  1.000000e+00  0.000000e+00   
25%    7.389000e+05  0.000000e+00  2.000000e+00  8.904000e+03  0.000000e+00   
50%    1.235059e+06  0.000000e+00  3.000000e+00  3.065200e+04  0.000000e+00   
75%    1.588160e+06  0.000000e+00  4.000000e+00  9.920400e+04  1.000000e+00   
max    1.715660e+06  1.000000e+00  5.000000e+00  1.783920e+05  1.000000e+00   

              score           time  
count  2.863265e+06  194821.000000  
mean   6.642891e+06     907.350630  
std    5.278561e+06     195.589391  
min    0.000000e+00      63.000000  
25%    9.302200e+05     719.000000  
50%    1.066027e+07     931.000000  
75%    1.141111e+07    1080.000000  
max    1.430034e+07    1200.000000

* 572,653 athlete entries from athlete dataset with 5 competition workouts per athlete results in 2,863,265 scores observations
* 13 features, 9 categorical & 4 numerical
* 14.7 % missing cells
* no duplicate rows
* memory size 1.2 GB

### feature explanation

In [13]:
df_19_sco.head()

affiliate                                          breakdown  \
0  CrossFit RDU                   9 rounds +\n10 wall-ball shots\n   
1  CrossFit RDU  Within 16 minutes:\n3 rounds +\n25 toes-to-bar...   
2  CrossFit RDU  200-ft. OH lunge\n50 box step-ups\n50 strict H...   
3  CrossFit RDU                                 132 reps\n6 rounds   
4  CrossFit RDU                                           210 reps   

   competitorid     division  is_scaled           judge  ordinal  rank  \
0         96511  Men (45-49)          0     Erin Miller        1     1   
1         96511  Men (45-49)          0  Harper Thorsen        2     4   
2         96511  Men (45-49)          0  Harper Thorsen        3     1   
3         96511  Men (45-49)          0  Harper Thorsen        4    36   
4         96511  Men (45-49)          0  Harper Thorsen        5     1   

   scaled     score scoredisplay       scoreidentifier   time  
0       0  13520000     352 reps  27f30f9a8c0a564ae799    NaN  
1       0  13420368     342 reps  0ed3d1264f25a8f1890d    NaN  
2       0  11800018         9:42  f2a143399a330c95321b  582.0  
3       0  11320009        11:51  89101e401c6c85997363  711.0  
4       0  12100573        10:27  f7588c9174f1fe90f5c4  627.0

* affiliate: the official name of an athlete's affiliate
* breakdown: contains information about the completed workout, it shows the amount of rounds or reps of every exercise and in addition the tiebreak time
* competitorid: the unique id of an athlete (also available in ahtlete dataset)
* division: the division (grouped by age) an athlete belongs to, same as in athlete dataset
* is_scaled: displays if an athlete belongs to a scaling division (overall), same as in athlete dataset
* judge: the person who judged the performance of an athlete for each workout
* ordinal: describes the workout number (1-5)
* rank: shows the rank of an athlete regarding one specific workout
* scaled: checks if the athleted performed a scaled version of a workout
* score: the score an athlete reached regarding one workout
* scoredisplay: shows the amount of total reps of a workout or the time when finished earlier than timecap
* scoreidentifier: ???
* time: if a workout was completed before timecap, time feature shows the time in seconds

### neglected features

* judge: name of the judge is not interesting
* scoreidentifier: seems to contain useless information